### PR TITLE
enhancement & fixes: update the code in preparation for handling SEP-31 non-happy-paths caused by `PENDING_CUSTOMER_INFO_UPDATE`

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Update the code in preparation for handling SEP-31 non-happy-paths caused by `PENDING_CUSTOMER_INFO_UPDATE`. The changes include updates in SEP-31-related classes. The most relevant changes are:
* **Update SEP-31 sequence diagram to preview the non-happy-path involving `pending_customer_info_update`** (not covering the implementation yet)
* Stop supporting `PENDING_TRANSACTION_INFO_UPDATE` in the TransactionEvent, to be compliant with the documentation.
* Add missing SEP-31 status `EXPIRED`.
* Fix PlatformAPI non-compliant field names.
* Add new check to SEP-12 to make sure the user status is ACCEPTED before proceeding with the SEP-31 tx creation.
* Fix Reference server’s `CustomerService` that was ignoring some customer info depending on the customer type.

Additionally, included some refactoring:
* Updates/fixes in the `Sep31Transaction` interface:
  * Add fields that were missing in the `Sep31Transaction` interface, but were present in the `JdbcSep31Transaction` class.
  * Move `Sep31Transaction.Refunds` and `Sep31Transaction.RefundPayments` to their own separate file.
* Updates in the `TransactionService`:
  * Rely on the `Sep31Transaction` interface instead of the `JdbcSep31Transaction` class.
  * Start supporting almost all fields
  * Add `TransactionServiceTest` class to test unit testing TransactionService (still only covering the GET request).
* Move `Sep31CustomerInfoNeededException` and `Sep31MissingFieldException` out of the Sep31Service class into their own file.
* Remove duplicated classes and move them to a shared folder: StellarId, Customers, Refund, RefundPayment.
* Add documentation to some methods in Sep31Service.

### Why

We'll need to rely on the PlatformAPI for the non-happy path and I had to make a bigger update to make sure it is compliant and ready for the new flow.

### CODE REVIEW

File: anchor-reference-server/src/test/kotlin/org/stellar/anchor/reference/service/UniqueAddressServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.stellar.anchor.reference.service;
```
Comment (lijamie98) [inline anchor-reference-server/src/test/kotlin/org/stellar/anchor/reference/service/UniqueAddressServiceTest.java:1]: The test is empty and can be removed.

I am not sure if we want to add unit tests to the reference server at this stage.

Comment (marcelosalloum) [reply anchor-reference-server/src/test/kotlin/org/stellar/anchor/reference/service/UniqueAddressServiceTest.java]: Thanks for surfacing that. FIxed in https://github.com/stellar/java-stellar-anchor-sdk/pull/440/commits/4e366835094714b123ee2a7b28198d00675bc3d3.

File: platform/src/test/kotlin/org/stellar/anchor/platform/service/TransactionServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,204 @@
+package org.stellar.anchor.platform.service
```
Comment (erika-sdf) [inline platform/src/test/kotlin/org/stellar/anchor/platform/service/TransactionServiceTest.kt:1]: Awesome, thanks for adding these tests!

File: docs/00 - Stellar Anchor Platform.md [hunk 3]
```diff
@@ -143,21 +143,49 @@ sequenceDiagram
...
     Stellar-->>Platform: receives payment, matches w/ transaction
-    Platform-->>Platform: updates transaction status
+    Platform-->>Platform: updates transaction status to `pending_receiver`
     Stellar-->>-Client: success response
     Platform->>+Anchor: POST [webhookURL]/transactions received
     Anchor-->>Anchor: queues off-chain payment
     Anchor-->>-Platform: 204 No Content
     Anchor->>Recipient: Sends off-chain payment to recipient
+
+    opt Mismatched info when delivering value to the Recipient (or a similar error)
```
Comment (erika-sdf) [inline docs/00 - Stellar Anchor Platform.md:167]: Thanks for updating the doc with this case

**Review by erika-sdf — APPROVED** [review]
Looks great, thanks for cleaning up so much while doing this!

**Pull request merged**